projects/plugins/boost/app/assets/src/js/pages/settings/sections/Modules.svelte

```diff
@@ -48,7 +48,19 @@
 
 	const suggestRegenerate = suggestRegenerateDS.store;
 
-	async function resume() {
+	$: if ( $modulesState.critical_css.available && $modulesState.critical_css.active ) {
+		resumeCriticalCssGeneration();
+	} else {
+		alreadyResumed = false;
```

Comment:
This still has the same problem as before - it's set in varying levels of abstraction - meaning that here it's **unset** directly in a Svelte expression, and it's **set** a level deeper - in a function. This makes the `resumeCriticalCssGeneration` function virtually useless because it doesn't accept any arguments, it just takes in (and modifies) values outside its scope, relying on the parent scope.

Reply:
If Critical CSS was running and you refreshed the page, it should be started only once (that's by onMount). Currently, it gets started by the active check as well (which is needed in case you toggle off and on so it can resume).

I can't see a way around using an external variable to monitor whether the function was already called or not. I'm also finding it difficult to figure out how to do that just by passing arguments to the function, without having an external variable.

Reply:
I ended up removing the function altogether. I also removed the onMount call since the "is module active" check calls it anyway.